python/lsst/summit/utils/peekExposure.py

```diff
@@ -0,0 +1,1026 @@
+                fratio = 1.234
+        # AuxTel/ComCam/LSSTCam all have 10 micron pixels (= 10e-3 mm)
+        donutDiameter = abs(focusZ) / fratio / 10e-3
+        self.log.info(f"{focusZ=} mm")
+        self.log.info(f"donutDiameter = {donutDiameter} pixels")
+        return donutDiameter
+
+    def run(
+        self,
+        exposure,
+        *,
+        doDisplay=False,
+        doDisplayIndices=False,
+        mode="auto",
+        donutDiameter=None,
+    ):
+        """
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure at which to peek.
+        doDisplay : bool, optional
+            Display the exposure and sources?  Default False.
+        doDisplayIndices : bool, optional
+            Display the source indices?  Default False.
+        mode : {'auto', 'donut', 'spec', 'photo'}, optional
+            Mode to run in.  Default 'auto'.
+        donutDiameter : float, optional
+            Donut diameter in pixels.  If None, will be estimated from exposure
+            metadata.  Default None.
+
+        Returns
+        -------
+        result : `pipeBase.Struct`
+            Result of the peek.
+            Struct containing:
+                - mode : str
+                    Peek mode that was run.
+                - binSize : int
+                    Binning factor used.
+                - binnedSourceCat : `lsst.afw.table.SourceCatalog`
+                    Source catalog from the binned exposure.
+                - table : `astropy.table.Table`
+                    Curated source table in unbinned coordinates.
+                - brightestIdx : int
+                    Index of the brightest source in the source catalog.
+                - brightestCentroid : `lsst.geom.Point2D`
+                    Centroid of the brightest source (unbinned coords).
+                - brightestShape : `lsst.geom.Quadrupole`
+                    Shape of the brightest source (unbinned coords).
+                - brightestShapeNW : `lsst.geom.Quadrupole`
+                    Shape of the brightest source in equitorial coordinates.
+                - brightestShapeAA : `lsst.geom.Quadrupole`
+                    Shape of the brightest source in alt/az coordinates.
+                - psfShape : `lsst.geom.Quadrupole`
+                    Estimated PSF shape (unbinned coords).
+                - psfShapeNW : `lsst.geom.Quadrupole`
+                    Estimated PSF shape in equitorial coordinates.
+                - psfShapeAA : `lsst.geom.Quadrupole`
+                    Estimated PSF shape in alt/az coordinates.
+        """
+        if donutDiameter is None:
+            donutDiameter = self.getDonutDiameter(exposure)
+
+        mode, binSize, binnedSourceCat = self.runPeek(
+            exposure, mode, donutDiameter
+        )
+
+        table = self.transformTable(binSize, binnedSourceCat)
+
+        wgood = self.getGoodSources(mode, binnedSourceCat)
```

Comment:
Maybe this is a convention I don't know of, but what's `w` here? _i.e._ why is this called `wgood`? I'd find it more readable if this was `goodSourceMask` or something like that.

Reply:
Haha.  This is channelling Josh from grad school where I extensively used the IDL `where` function and lazily set the return value to `w`.  Happy to use `goodSourceMask`.